apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/AggregatesServiceDefault.java

```diff
@@ -0,0 +1,179 @@
+import it.pagopa.selfcare.onboarding.model.AggregatesCsvResponse;
+import it.pagopa.selfcare.onboarding.model.CsvAggregate;
+import it.pagopa.selfcare.onboarding.model.RowError;
+import it.pagopa.selfcare.onboarding.model.VerifyAggregateResponse;
+import it.pagopa.selfcare.onboarding.util.InstitutionPaSubunitType;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.WebApplicationException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.logging.Logger;
+import org.openapi.quarkus.party_registry_proxy_json.api.AooApi;
+import org.openapi.quarkus.party_registry_proxy_json.api.InstitutionApi;
+import org.openapi.quarkus.party_registry_proxy_json.api.UoApi;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.opencsv.ICSVParser.DEFAULT_QUOTE_CHARACTER;
+import static io.vertx.core.http.impl.HttpClientConnection.log;
+
+@ApplicationScoped
+@Slf4j
+public class AggregatesServiceDefault implements AggregatesService{
+
+    private static final Logger LOG = Logger.getLogger(AggregatesServiceDefault.class);
+
+    @Inject
+    OnboardingMapper onboardingMapper;
+
+    @RestClient
+    @Inject
+    AooApi aooApi;
+
+    @RestClient
+    @Inject
+    InstitutionApi institutionApi;
+
+    @RestClient
+    @Inject
+    UoApi uoApi;
+
+    public static final String ERROR_READING_CSV = "Error reading CSV: ";
+    public static final String MALFORMED_ROW = "Riga malformata";
+    public static final String ERROR_IPA = "Codice fiscale non presente su IPA";
+    public static final String ERROR_TAXCODE = "Il codice fiscale è obbligatorio";
+    public static final String ERROR_DESCRIPTION = "La ragione sociale è obbligatoria";
+
+    @Override
+    public Uni<VerifyAggregateResponse> validateAggregatesCsv(File file){
+        AggregatesCsvResponse aggregatesCsvResponse = readItemsFromCsv(file);
+        List<CsvAggregate> csvAggregates = aggregatesCsvResponse.getCsvAggregateList();
+        return Multi.createFrom().iterable(csvAggregates)
+                .onItem().transformToUniAndMerge(csvAggregate -> checkCsvAggregateAndFillErrorList(csvAggregate, aggregatesCsvResponse))
+                .collect().asList()
+                .onItem().transform(list -> onboardingMapper.toVerifyAggregateResponse(aggregatesCsvResponse))
+                .onItem().invoke(() -> LOG.infof("CSV file validated end: %s valid row and %s invalid row",
+                        aggregatesCsvResponse.getValidAggregates().size(),
+                        aggregatesCsvResponse.getRowErrorList().size()));
+    }
+
+    private Uni<Void> checkCsvAggregateAndFillErrorList(CsvAggregate csvAggregate, AggregatesCsvResponse aggregatesCsvResponse) {
```

Comment:
you could add to *FillAggregate* on method signature....for ex. checkCsvAggregateAndFillAggregateOrErrorList